src/expr.rs

```diff
@@ -2125,21 +2142,23 @@ pub(crate) fn rewrite_assign_rhs_with_comments<S: Into<String>, R: Rewrite>(
     rhs_tactics: RhsTactics,
     between_span: Span,
     allow_extend: bool,
-) -> Option<String> {
+) -> RewriteResult {
     let lhs = lhs.into();
     let contains_comment = contains_comment(context.snippet(between_span));
     let shape = if contains_comment {
-        shape.block_left(context.config.tab_spaces())?
+        shape
+            .block_left(context.config.tab_spaces())
+            .max_width_error(shape.width, between_span.with_hi(ex.span().hi()))?
```

Comment:
```rust
... = // comment 
       Expr
```
Since `between_span` corresponds to the span of `// comment` but we are calculating the shape for the `expr`, I think the span for `max_width_error` should start from `between_span.lo()` and end with `expr.span.hi()`